### ISSUE TITLE
fusion: CrossAttentionFusion is incompatible with marginalized_retrieval — add PerDocCrossAttentionFusion

### ISSUE THREAD

**Issue by stepanyanhayk** [McDermottHealthAI/MedRAP#64]
## Problem

`CrossAttentionFusion.fuse` returns `fused_state` shaped `(B, S_ehr, d_model)` — a fused patient sequence. When paired with `marginalized_retrieval=True` in `RetrievalAugmentedModel`, the `ndim == 3` guard passes silently but the semantics are wrong:

```python
b, k_docs, d_mem = fused.shape   # k_docs is actually S_ehr, not K
per_doc_logits = self.head(fused.reshape(-1, d_mem)).view(b, k_docs, num_classes)
```

The model marginalizes over **EHR sequence positions** instead of **retrieved documents**. There is no error, just silently incorrect predictions and gradients.

## What is needed

A cross-attention fusion variant that produces `fused_state` shaped `(B, K, d_model)` — one enriched vector per retrieved document — so the marginalized retrieval path in `model.py` operates over the correct K dimension.

The design: expand the batch to `B*K` (one element per patient-doc pair), run cross-attention between each patient and its corresponding single document, mean-pool over the patient sequence, then reshape to `(B, K, d_model)`.

## Relationship to existing issues

- Part of the broader fusion/marginalized-retrieval compatibility gap noted in #55.
- The `(B, K, d_model)` output shape requirement is set by `model.py`'s `marginalized_retrieval=True` path.
